File: Interpreter/CosFun.java
package calculator.xwg;

import java.util.LinkedList;

public class CosFun extends CalculateFunction {
	String getName(){
		return "cos";
	}
	
	boolean execute(LinkedList<Complex> paraList, EvaluateContext context){
		if(paraList.size() != 1)
	    {
	        context.setErrorMessage(getName(), R.string.error_invalid_parameter_count);
	        return false;
	    }
	    Complex para = paraList.getFirst();
	    if(para.i != 0)
	    {
	        context.setErrorMessage(getName(), R.string.error_invalid_date_type);
	        return false;
	    }
	    double result = StrictMath.cos(para.r);
	    context.setCurrentResult(new Complex(result));
	    return true;
	}
}

File: Interpreter/CosFun.py
from Interpreter.CalculateFunction import *
from Interpreter.Complex import *


class CosFun (CalculateFunction):
	def get_name(self):
		return 'cos'

	def execute(self, context, paraList):
		if len(paraList) != 1:
			context.setErrorMessage(self.get_name(), R_string.error_invalid_parameter_count)
			return False
		para = paraList[0]
		if para.i:
			context.setErrorMessage(self.get_name(), R_string.error_invalid_date_type)
			return False
		result = math.cos(para.r)
		context.setCurrentResult(Complex(result))
		return True
